app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -77,10 +73,58 @@ public static SessionStore get() {
         GeckoSession mSession;
     }
 
+    private static class GeckoError {
+
+        public static final String ERROR_URL = "resource://android/assets/html/error/netError.xhtml";
+
+        public static final String NET_ERROR_TYPE = "about:neterror";
+        public static final String CERT_ERROR_TYPE = "about:certerror";
+        public static final String BLOCKED_ERROR_TYPE = "about:blocked";
```

Comment:
**malicious sites** do not appear to be blocked. and the document doesn't change but the URL does, which is confusing/incorrect.

example: http://malware.wicar.org/data/eicar.com:

![malware wicard org eicar 0](https://user-images.githubusercontent.com/203725/43291513-885c2e9a-90e6-11e8-99a8-8ba0d9cc0056.png)
![image](https://user-images.githubusercontent.com/203725/43291550-a97915c0-90e6-11e8-9b3e-a57b26b40bc9.png)
![image](https://user-images.githubusercontent.com/203725/43291552-aaf6d504-90e6-11e8-9529-b4014a32d12f.png)

Reply:
Also seems to be related to Gecko, we get a onPageStart and onPageStop but the rendered content doesn't change. Android Firefox also doesn't show the Malware error page it just downloads the file.